traffic_ops/traffic_ops_golang/capabilities/capabilities.go

```diff
@@ -0,0 +1,305 @@
+	var err error
+	if name, ok := inf.Params["name"]; ok {
+		rows, err = tx.Query(readQuery + "WHERE name=$1", name)
+	} else {
+		rows, err = tx.Query(readQuery)
+	}
+	if err != nil && err != sql.ErrNoRows {
+		errCode = http.StatusInternalServerError
+		sysErr = fmt.Errorf("querying capabilities: %v", err)
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+	defer rows.Close()
+
+	caps := []tc.Capability{}
+	for rows.Next() {
+		cap := tc.Capability{}
+		if err := rows.Scan(&cap.Description, &cap.LastUpdated, &cap.Name); err != nil {
+			errCode = http.StatusInternalServerError
+			sysErr = fmt.Errorf("Parsing database response: %v", err)
+			api.HandleErr(w, r, tx, errCode, nil, sysErr)
+			return
+		}
+
+		caps = append(caps, cap)
+	}
+
+	respBts, err := json.Marshal(struct{R []tc.Capability `json:"response"`}{caps})
+	if err != nil {
+		errCode = http.StatusInternalServerError
+		sysErr = fmt.Errorf("Marshaling response: %v", err)
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+
+	w.Header().Set(tc.ContentType, tc.ApplicationJson)
+	w.Write(append(respBts, '\n'))
+}
+
+func Create(w http.ResponseWriter, r *http.Request) {
+	inf, sysErr, userErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	decoder := json.NewDecoder(r.Body)
+	var cap tc.Capability
+	if err := decoder.Decode(&cap); err != nil {
+		sysErr = fmt.Errorf("Decoding request body: %v")
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+
+	if cap.Name == "" {
+		userErr = errors.New("'name' must be defined! (and not empty)")
+		errCode = http.StatusBadRequest
+		api.HandleErr(w, r, tx, errCode, userErr, nil)
+		return
+	}
+
+	if cap.Description == "" {
+		userErr = errors.New("'description' must be defined! (and not empty)")
+		errCode = http.StatusBadRequest
+		api.HandleErr(w, r, tx, errCode, userErr, nil)
+		return
+	}
+
+	if ok, err := capabilityNameExists(cap.Name, tx); err != nil {
+		sysErr = fmt.Errorf("Checking for capability %s's existence: %v", cap.Name, err)
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	} else if ok {
+		userErr = fmt.Errorf("Capability '%s' already exists!", cap.Name)
+		errCode = http.StatusConflict
+		api.HandleErr(w, r, tx, errCode, userErr, nil)
+		return
+	}
+
+	row := tx.QueryRow(createQuery, cap.Name, cap.Description)
+	if err := row.Scan(&cap.Description, &cap.LastUpdated, &cap.Name); err != nil {
+		sysErr = fmt.Errorf("Inserting capability: %v", err)
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, tx, errCode, nil, sysErr)
+		return
+	}
+
+	alert := []tc.Alert{tc.Alert{Level: tc.SuccessLevel.String(), Text: "Capability created."}}
+	responseObj := struct{A []tc.Alert `json:"alerts"`; R tc.Capability `json:"response"`}{alert, cap}
```

Comment:
same comment here I think there is an API util function that does this as well as the other handlers below